### DISCUSSION TITLE
What is the future of @ngrx/data?. Really worry about it.

### DISCUSSION THREAD

**Discussion by estebanluengosimon** [ngrx/platform#4212] [General]
Hi guys,

We've recently started to migrate our projects to Angular v17. We came across some different discussions about the future of @ngrx/data in this [link](https://github.com/ngrx/platform/issues/4011). 
Firstly, it seems as if you're planning to maybe deprecated this package or perhaps not continuing adding more functionality and letting die by itself. It's really confusing at this point.
On the other hand, we don't have a clear idea how to migrate a module based app using @ngrx/data to a standalone app. For instance, What is the best way to register data services using EntityDatServices?. The doc is based only in module approch. 
So, it's important for my company knowing the future of this package as we have a considerable number of apps based on this package. For us it was a marvelous help using @ngrx/data as we could reduce a considerable amount of boiler plate. We have thousands of data tables and seven backend apps that manage these data from a frontend writing in Angular using @ngrx/data.
In case you are planning to recommend other alternatives such as signalStore, it'll be a great help for us if you could describe some doc about how to migrate taking into account that out there there are companies that have thousands of lines of code using this technology. 

Thanks a lot.

**Comment by david-shortman** [comment]
I'd predict that the switch to maintenance mode won't be a concern for your apps for years, allowing for a few different paths.

The entity package contains relatively simple functions that haven't need to be touched for years because of their straightforward nature. In a doomsday scenario where the package immediately ended support, you could copy the library code into your repository or create an internal npm package and it would be likely to continue working for years.

There is a guide for using Entities in the SignalStore here- https://ngrx.io/guide/signals/signal-store/entity-management
